pkg/controller/actions/architecture/action_verify_supported_architectures.go

```diff
@@ -0,0 +1,82 @@
+package architecture
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odherrors "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/errors"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/conditions"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+)
+
+// VerifySupportedArchitectures determines whether a component can be enabled based on the architecture of each node.
+//
+// This is accomplished by doing the following:
+// 1. Fetching the architecture(s) that the component supports.
+// 2. Fetching the architecture(s) that the nodes are running on.
+// 3. Verifying that there is at least one node that is running an architecture supported by the component.
+func VerifySupportedArchitectures(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	obj, ok := rr.Instance.(common.WithReleases)
+	if !ok {
+		return fmt.Errorf("instance %v is not a WithReleases", rr.Instance)
+	}
+
+	// Fetch the architecture(s) that the component supports
+	supportedArchitectures := make(map[string]struct{})
+	componentReleases := obj.GetReleaseStatus()
+	if componentReleases == nil || len(*componentReleases) < 1 {
+		return fmt.Errorf("instance %v has no releases", rr.Instance)
+	}
+	for _, release := range *componentReleases {
+		for _, arch := range release.SupportedArchitectures {
+			supportedArchitectures[arch] = struct{}{}
+		}
+	}
+
+	// TODO: Refactor after all components explicitly list supportedArchitectures in their component_metadata.yaml file
+	// If supportedArchitectures is empty, assume the component only works on amd64
+	if len(supportedArchitectures) == 0 {
+		supportedArchitectures["amd64"] = struct{}{}
```

Comment:
maybe `amd64` should be defined as a constant, like `DefaultArchitecture`

Reply:
Good catch, `amd64` should definitely be a constant. I fixed this in the latest commit.